src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -308,30 +313,92 @@ def _deploy_single_object(
+    index_url: Optional[str] = IndexUrlOption,
+    skip_version_check: bool = SkipVersionCheckOption,
+    deprecated_package_native_libraries: YesNoAsk = deprecated_allow_native_libraries_option(
+        "--package-native-libraries"
+    ),
+    deprecated_check_anaconda_for_pypi_deps: bool = DeprecatedCheckAnacondaForPyPiDependencies,
+    _deprecated_pypi_download: YesNoAsk = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Builds the Snowpark project as a `.zip` archive that can be used by `deploy` command.
     The archive is built using only the `src` directory specified in the project file.
     """
+    if not deprecated_check_anaconda_for_pypi_deps:
+        ignore_anaconda = True
+    # TODO: yes/no/ask logic should be removed in 3.0
+    allow_shared_libraries_yesnoask = {
+        True: YesNoAsk.YES,
+        False: YesNoAsk.NO,
+    }[allow_shared_libraries]
+    if deprecated_package_native_libraries != YesNoAsk.NO:
+        allow_shared_libraries_yesnoask = deprecated_package_native_libraries
+
     paths = SnowparkPackagePaths.for_snowpark_project(
         project_root=SecurePath(cli_context.project_root),
         snowpark_project_definition=cli_context.project_definition,
     )
     log.info("Building package using sources from: %s", paths.source.path)
 
-    snowpark_package(
-        paths=paths,
-        pypi_download=pypi_download,  # type: ignore[arg-type]
-        check_anaconda_for_pypi_deps=check_anaconda_for_pypi_deps,
-        package_native_libraries=package_native_libraries,  # type: ignore[arg-type]
-    )
+    with SecurePath.temporary_directory() as packages_dir:
+        if paths.defined_requirements_file.exists():
+            log.info("Resolving any requirements from requirements.txt...")
+            requirements = package_utils.parse_requirements(
+                requirements_file=paths.defined_requirements_file,
+            )
+
+            download_result = package_utils.download_unavailable_packages(
+                requirements=requirements,
+                target_dir=packages_dir,
+                ignore_anaconda=ignore_anaconda,
+                skip_version_check=skip_version_check,
+                pip_index_url=index_url,
+                allow_shared_libraries=allow_shared_libraries_yesnoask,
+            )
+            if not download_result.download_successful:
+                return MessageResult(download_result.error_message)
+            if download_result.packages_available_in_anaconda:
+                _write_requirements_file(
+                    paths.snowflake_requirements_file,
+                    download_result.packages_available_in_anaconda,
+                )
+
+        zip_dir(source=paths.source.path, dest_zip=paths.artifact_file.path)
+        if any(packages_dir.iterdir()):
+            zip_dir(
+                source=packages_dir.path,
+                dest_zip=paths.artifact_file.path,
+                mode="a",
```

Comment:
why append mode?

Reply:
Creating .zip artifact happens in two steps:
```
pack_code
if are_there_any_packages
  pack_packages(in_append_mode)
  ```

Reply:
Nice! Let's add this comment in code

Reply:
done